parla/task_runtime.py

```diff
@@ -1126,48 +1123,45 @@ def check_resources_availability(self, d: Device, resources: ResourceDict):
         :param resources: The resources to deallocate.
         """
         logger.debug("[ResourcePool] Acquiring monitor in check_resources_availability()")
-        with self._monitor:
+        is_available = True
+        for name, amount in resources.items():
+            dres = self._devices[d]
 
-            is_available = True
-            for name, amount in resources.items():
-                dres = self._devices[d]
+            # Workaround stupid vcus (I'm getting rid of these at some point)
+            #if d.architecture.id == 'gpu' and name == 'vcus':
+            #    continue
 
-                # Workaround stupid vcus (I'm getting rid of these at some point)
-                #if d.architecture.id == 'gpu' and name == 'vcus':
-                #    continue
-
-                if amount > dres[name]:
-                    is_available = False
-                logger.debug("Resource check for %d %s on device %r: %s", amount, name, d, "Passed" if is_available else "Failed")
-            logger.debug("[ResourcePool] Releasing monitor in check_resources_availability()")
-            return is_available
+            if amount > dres[name]:
+                is_available = False
+            logger.debug("Resource check for %d %s on device %r: %s", amount, name, d, "Passed" if is_available else "Failed")
+        logger.debug("[ResourcePool] Releasing monitor in check_resources_availability()")
+        return is_available
 
     def _atomically_update_resources(self, d: Device, resources: ResourceDict, multiplier, block: bool):
         logger.debug("[ResourcePool] Acquiring monitor in atomically_update_resources()")
-        with self._monitor:
```

Comment:
same question, are we sure this is safe to remove? Is it because only the scheduler calls this now?